### PR TITLE
feat: ancestor & items-control bindings

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): n/a

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Added AncestorBinding + ItemsControlsBinding

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information
ninja fix: design-agnotic control dont have a proper page header && drawer-control sample broken with the m2/3 update (due to SamplePageLayout)

### CODE REVIEW

File: doc/helpers/ancestor-itemscontrol-binding.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# AncestorBinding & ItemsControlBinding
+These markup extensions provides relative binding based on ancestor type. If you are familiar with WPF, they are very similar to `{RelativeSource Mode=FindAncestor}`.
+
+## Remarks
+Both of the markup extensions rely on a WinUI 3 to work. It is not available on UWP for windows. You can however still use it on uno platforms with UWP.
```
Comment (Xiaoy312) [inline doc/helpers/ancestor-itemscontrol-binding.md:5]: not sure how i can better word this..? 

Target|UWP|WINUI
-|-|-
win|not-ok|ok
uno|ok|ok

File: src/Uno.Toolkit.UI/Markup/ItemsControlBindingExtension.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+using Windows.UI.Xaml.Media;
+#endif
+
+namespace Uno.Toolkit.UI
+{
+	/// <summary>
+	/// Provides a binding to the closest parent ItemsControl.
+	/// This markup can be used to access the parent ItemsControl from inside of the ItemTemplate.
+	/// </summary>
+	public class ItemsControlBindingExtension : MarkupExtension
```
Comment (jeromelaban) [inline src/Uno.Toolkit.UI/Markup/ItemsControlBindingExtension.cs:26]: The code for this one looks like a duplicate of the `AncestorBindingExtension`, could we aggregate/inherit from it?

File: doc/helpers/ancestor-itemscontrol-binding.md (hunk 1 above)
Comment (kazo0) [inline doc/helpers/ancestor-itemscontrol-binding.md:5]:
```suggestion
Both of the markup extensions are available for all **non-Windows** UWP platforms as well as all WinUI 3 platforms. 
```

Still sounds kind of confusing though, the table might make it more clear

**Review by jeromelaban — APPROVED** [review]
🚀

**Review by kazo0 — APPROVED** [review]

File: doc/helpers/ancestor-itemscontrol-binding.md (hunk 1 above)
Comment (jeromelaban) [reply doc/helpers/ancestor-itemscontrol-binding.md]: Let's see how developers read this, we'll adjust as needed.

**Pull request merged**